embedded-graphics/src/primitives/triangle.rs

```diff
@@ -61,44 +60,22 @@ impl<C> Dimensions for Triangle<C>
 where
     C: PixelColor,
 {
-    fn top_left(&self) -> Coord {
-        let x = if self.p1[0] <= self.p2[0] && self.p1[0] <= self.p3[0] {
-            self.p1[0]
-        } else if self.p2[0] <= self.p1[0] && self.p2[0] <= self.p3[0] {
-            self.p2[0]
-        } else {
-            self.p3[0]
-        };
-        let y = if self.p1[1] <= self.p2[1] && self.p1[1] <= self.p3[1] {
-            self.p1[1]
-        } else if self.p2[1] <= self.p1[1] && self.p2[1] <= self.p3[1] {
-            self.p2[1]
-        } else {
-            self.p3[1]
-        };
-        Coord::new(x, y)
+    fn top_left(&self) -> Point {
+        let &x = [self.p1.x, self.p2.x, self.p3.x].iter().min().unwrap();
+        let &y = [self.p1.y, self.p2.y, self.p3.y].iter().min().unwrap();
```

Comment:
Thanks for making this code much nicer!

![image](https://user-images.githubusercontent.com/800791/63895373-e13c5780-c9e6-11e9-95f6-85cef6e52034.png)